File: lessons/compiled_lang/cs/src/f40.loop.cs
// testbox: title="while (true) with continue"
using System;

class F40Loop
{
    static void Main()
    {
        while (true)
        {
            Console.Write("Enter your name (quit to exit): ");
            string answer = Console.ReadLine();

            if (string.IsNullOrWhiteSpace(answer))
                continue;

            if (answer == "quit")
                break;

            Console.WriteLine($"Hello {answer}!");
        }
    }
}

File: lessons/compiled_lang/rust/src/f40.loop.rs
// testbox: title="loop {} with continue"
use std::io::{self, Write};

fn main() {
    loop {
        print!("Enter your name (quit to exit): ");
        io::stdout().flush().unwrap();

        let mut line = String::new();
        io::stdin().read_line(&mut line).unwrap();
        let answer = line.trim_end().to_string();

        if answer.trim().is_empty() {
            continue;
        }

        if answer == "quit" {
            break;
        }

        println!("Hello {}!", answer);
    }
}
